src/regex/lib_pcre.cr

```diff
@@ -23,5 +23,7 @@ lib LibPCRE
   $pcre_free : Free
 end
 
-LibPCRE.pcre_malloc = ->GC.malloc(LibC::SizeT)
-LibPCRE.pcre_free = ->GC.free(Void*)
+{% unless flag?(:without_pcre) %}
```

Comment:
The macro idea is good, we sometimes need to run code but only if the lib is actually used. The "problem" is that the compiler finds out if a lib is used when a lib fun is invoked, and at that point the macro would need to be expanded and typed... and then the codegen should put that code right where the macro definition exists. I'll try to think how to implement this, maybe it's not that hard. I'm thinking about something like `macro initialize`, but another (better) name should be good too.